pkg/java/pom/pom.go

```diff
@@ -253,10 +254,12 @@ func (d pomDependency) Resolve(props map[string]string, depManagement, rootDepMa
 
 // ToArtifact converts dependency to artifact.
 // It should be called after calling Resolve() so that variables can be evaluated.
-func (d pomDependency) ToArtifact(exclusions map[string]struct{}) artifact {
-	if exclusions == nil {
-		exclusions = map[string]struct{}{}
-	}
+func (d pomDependency) ToArtifact(ex map[string]struct{}) artifact {
+	// To avoid shadow adding exclusions to top pom's,
+	// we need to initialize a new map for each new artifact
+	// See `exclusions in child` test for more information
+	exclusions := map[string]struct{}{}
+	maps.Copy(exclusions, ex)
```

Comment:
How about maps.Clone?

```suggestion
	exclusions := maps.Clone(ex)
```

Reply:
Looks like we will have enough of a hallow clone. Thank you!
Changed in https://github.com/aquasecurity/go-dep-parser/pull/282/commits/cdc2c7f5f7e7e092457c0c18762b35339495b7cf